roles/get-sitename/tasks/main.yml

```diff
@@ -53,28 +53,66 @@
   when:
     acsf_environment != ""
 
-- name: Get list of sites on Site Factory
-  uri:
-    url: "https://www.{{ sitefactory_environment }}cardinalsites.acsitefactory.com/api/v1/sites?limit=100"
-    method: GET
-    user: "{{ acquia_username }}"
-    password: "{{ acquia_api_key }}"
-    force_basic_auth: yes
-    return_content: yes
-  register: existing_sites
+# - name: Get list of sites on Site Factory
+#   uri:
+#     url: "https://www.{{ sitefactory_environment }}cardinalsites.acsitefactory.com/api/v1/sites?limit=100"
+#     method: GET
+#     user: "{{ acquia_username }}"
+#     password: "{{ acquia_api_key }}"
+#     force_basic_auth: yes
+#     return_content: yes
+#   register: existing_sites
 
-#- name: Display all site names
-#  debug:
-#    var: item
-#  with_items: "{{ existing_sites | json_query('json.sites[*].site') }}"
+- name: Get sites.json from remote
+  command: drush @acsf{{ drush_environment }}.cardinald7.saml php-eval "include 'sites/sites.php'; drupal_json_output(gardens_site_data_load_file());"
+  register: sites_json_raw
 
-- name: Establish numerical id from existing_sites JSON response
+- name: Get sites.json from remote
   set_fact:
-    site_id: "{{ existing_sites | json_query(existing_sites_query) }}"
+    sites_json: "{{ sites_json_raw.stdout|from_json }}"
+
+# - name: debug sites.json
+#   debug:
+#     var: sites_json
+
+#
+# Creates a format that looks like:
+# {
+#     "id": 5001626,
+#     "site": "crceducation",
+#     "url": "crceducation-dev.cardinalsites.stanford.edu"
+# },
+#
+- name: Manipulate sites.json
+  set_fact:
+    sites_formatted: "{{ sites_formatted|default([]) + [ {'url': item.0, 'id': item.1.acsf_site_id, 'site': item.0|regex_replace('((-dev|-stage|-test).cardinalsites.stanford.edu|.(dev-|stage-|test-)cardinalsites.acsitefactory.com|(-dev|-stage|-test).sites.stanford.edu)', '') } ] }}"
+  with_together:
+    - "{{ sites_json.sites }}"
+    - "{{ sites_json|json_query('sites.*.conf') }}"
+
+# - name: debug sites_formatted
+#   debug:
+#     var: sites_formatted
+
+- name: Establish numerical id from sites_formatted JSON response
+  set_fact:
+    possible_site_ids: "{{ sites_formatted|json_query(existing_sites_query) }}"
   vars:
-    existing_sites_query: "json.sites[?site=='{{ acsf_site_name }}'] | [0].id"
+    existing_sites_query: "[?site=='{{ acsf_site_name }}'].id"
 
-#- name: Debug json formatting on sites_created id
-#  debug:
-#    var: site_id
+# - name: debug possible_site_ids
+#   debug:
+#     var: possible_site_ids
 
+- name: Set site_id if site id exists
+  set_fact:
+    site_id: "{{ possible_site_ids|first }}"
```

Comment:
The possible site ids at this point should all be the same. They may have been out of order as sites.stanford.edu was added to the list of possible domains that could be used during migration. I have added that domain to line 88 as it is super important that the site key in the sites_formatted collection be the key in which is in the inventory playbook.

Reply:
Hmm... that may be where the logic fails... I am assuming that the domains match the key in this instance but we already have custom vanities that could produce a false positive. I will have to limit the results to the acsitefactory url to ensure that it really is a key.